pkg/loadbalancer/roundrobin.go

```diff
@@ -85,6 +82,39 @@ func (rr *roundRobin) GetRemoteNodeConnection(ctx context.Context) (*grpc.Client
 
 	// Get target node info and set next round robbin node.
 	// nextNode is always lastNode + 1 mod (numOfNodes), to loop back to zero
+	targetNodeEndpoint, isRemoteConn := rr.getTargetAndIncrement(&cluster)
```

Comment:
Do you think we can remove the `rr.cleanupMissingNodeConnections()` call from line 81? Anyway we are doing that as part of the 15 min periodic cleanup. This will reduce some more processing time from this hot GetRemoteNodeConnection function?

Reply:
Yeah, we could do that. However, the majority of the time spent is likely on the Enumerate and Connection create. `cleanupMissingNodeConnections` right here pretty much guarantees that a remote connection will be to a valid node.

I could see a case where one removed node would cause round robin to halt on a dead node.

We could skip over dead nodes by incrementing the counter and retrying, but I think that would be slightly more involved change.

Reply:
This could be optimized further by removing cluster.Enumerate in this hot path altogether in favor of a node watcher+cache in a separate thread, but we can save that for another time.